### ISSUE TITLE
Understand if it is required to rebase icub3/wbd on robotology/icub-models/devel

### ISSUE THREAD

**Issue by GiulioRomualdi** [ami-iit/icub-models#1]
Before starting using the model of `iCubGazeboV3` I ran a `git diff` between this model and the one stored in ` robotology/icub-models`. Namelly I compared:
- 20d034c547aae832c7f3db3c1f53398dc10fac76: latest `icub3/wbd` commit of this fork
- https://github.com/robotology/icub-models/commit/3c6cd409db01c4f3883583acb45525f56a2e52c4: latest devel of `robotology/icub-models`

`git diff 20d034c547aae832c7f3db3c1f53398dc10fac76..3c6cd409db01c4f3883583acb45525f56a2e52c4  iCub/robots/iCubGazeboV3/model.urdf`

this is the output. 
There seems there is a difference between some inertial and more important the direction of some axis, namely `r_hip_yaw` and `r_hip_roll`. If I've to tell the truth I don't know if this modification is required. In  https://github.com/robotology/icub-models-generator/pull/153 I modified some axis direction but it's strange that it has been ported on the model only now

```patch
diff --git a/iCub/robots/iCubGazeboV3/model.urdf b/iCub/robots/iCubGazeboV3/model.urdf
index 11cc82f..40ea752 100644
--- a/iCub/robots/iCubGazeboV3/model.urdf
+++ b/iCub/robots/iCubGazeboV3/model.urdf
@@ -75,7 +75,7 @@
   </link>
   <joint name="r_hip_roll" type="revolute">
     <origin xyz="-0.0473 -0.03310000000000001 -0.009000000000000008" rpy="0 0 0"/>
-    <axis xyz="1.0 0.0 0.0"/>
+    <axis xyz="-1.0 0.0 0.0"/>
     <parent link="r_hip_1"/>
     <child link="r_hip_2"/>
     <limit effort="50000" lower="-0.17453292519943295" upper="2.007128639793479" velocity="120.0"/>
@@ -133,7 +133,7 @@
   </link>
   <joint name="r_hip_yaw" type="revolute">
     <origin xyz="0 0.015000000000000006 -0.07090000000000007" rpy="0 0 0"/>
-    <axis xyz="0.0 0.0 1.0"/>
+    <axis xyz="0.0 0.0 -1.0"/>
     <parent link="r_hip_3"/>
     <child link="r_upper_leg"/>
     <limit effort="50000" lower="-1.3962634015954636" upper="1.3962634015954636" velocity="120.0"/>
@@ -173,7 +173,7 @@
     <inertial>
       <origin xyz="-0.0052558000000000014 0.002301599999999987 -0.018286000000000024" rpy="0 0 -1.5707963267948966"/>
       <mass value="1.80595"/>
-      <inertia ixx="0.00329109" ixy="-0.00011193325094648436" ixz="-0.00023180543265495062" iyy="0.00097845" iyz="-0.00010678991458219672" izz="0.00298837"/>
+      <inertia ixx="0.01" ixy="-0.00011193325094648436" ixz="-0.00023180543265495062" iyy="0.01" iyz="-0.00010678991458219672" izz="0.01"/>
     </inertial>
     <visual>
       <origin xyz="-0.02600000000000011 0.08304999999996568 0.8456470000000077" rpy="0 0 -1.5707963267948966"/>
@@ -203,7 +203,7 @@
     <inertial>
       <origin xyz="0.0435896 9.779999999998124e-05 -0.02621299999999993" rpy="0 0 -1.5707963267948966"/>
       <mass value="1.06976"/>
-      <inertia ixx="0.00510275" ixy="2.566239505873549e-06" ixz="-4.1923537473188265e-06" iyy="0.000995114" iyz="0.000794242494242394" izz="0.00458166"/>
+      <inertia ixx="0.01" ixy="2.566239505873549e-06" ixz="-4.1923537473188265e-06" iyy="0.01" iyz="0.000794242494242394" izz="0.01"/>
     </inertial>
     <visual>
       <origin xyz="0.026749999999999875 0.0648499999999391 0.8806470000000108" rpy="0 0 -1.5707963267948966"/>
@@ -223,7 +223,7 @@
   </link>
   <joint name="r_ankle_roll" type="revolute">
     <origin xyz="-0.052750000000000005 0.018200000000000008 -0.03500000000000003" rpy="0 0 0"/>
-    <axis xyz="1.0 0.0 0.0"/>
+    <axis xyz="-1.0 0.0 0.0"/>
     <parent link="r_ankle_1"/>
     <child link="r_ankle_2"/>
     <limit effort="50000" lower="-0.4363323129985824" upper="0.4363323129985824" velocity="120.0"/>
@@ -232,8 +232,8 @@
   <link name="r_foot_front">
     <inertial>
       <origin xyz="0.0005860000000000032 0 0.0029439999999999467" rpy="0 0 -1.5707963267948966"/>
-      <mass value="0.723938"/>
-      <inertia ixx="0.000770637" ixy="0.0" ixz="0.0" iyy="0.000568064" iyz="-2.3758437778657626e-08" izz="0.001333"/>
+      <mass value="0.184375"/>
+      <inertia ixx="0.01" ixy="0.0" ixz="0.0" iyy="0.01" iyz="-6.05088e-09" izz="0.01"/>
     </inertial>
     <visual>
       <origin xyz="-0.10850000000000011 0.06484999999993908 0.9464470000000105" rpy="0 0 -1.5707963267948966"/>
@@ -245,9 +245,9 @@
       </material>
     </visual>
     <collision>
-      <origin xyz="-0.10850000000000011 0.06484999999993908 0.9464470000000105" rpy="0 0 -1.5707963267948966"/>
+      <origin xyz="0.0 0.0 0.003" rpy="0.0 0.0 0.0"/>
       <geometry>
-        <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_r_foot_front_prt.stl" scale="0.001 0.001 0.001"/>
+        <box size="0.117 0.1 0.006"/>
       </geometry>
     </collision>
   </link>
@@ -260,8 +260,8 @@
   <link name="r_foot_rear">
     <inertial>
       <origin xyz="-0.00021610000000000032 -9.280000000001787e-05 0.00317400000000001" rpy="0 0 -1.5707963267948966"/>
-      <mass value="0.728095"/>
-      <inertia ixx="0.000769193" ixy="-1.0104094827945897e-06" ixz="-8.724056225297856e-08" iyy="0.000565835" iyz="7.812644807512598e-08" izz="0.00132907"/>
+      <mass value="0.18484137"/>
+      <inertia ixx="0.01" ixy="-2.5651250600641865e-07" ixz="-2.2147748640508236e-08" iyy="0.01" iyz="1.983394982171303e-08" izz="0.01"/>
     </inertial>
     <visual>
       <origin xyz="0.010749999999999881 0.0648499999999391 0.9464470000000105" rpy="0 0 -1.5707963267948966"/>
@@ -273,9 +273,9 @@
       </material>
     </visual>
     <collision>
-      <origin xyz="0.010749999999999881 0.0648499999999391 0.9464470000000105" rpy="0 0 -1.5707963267948966"/>
+      <origin xyz="0.0 0.0 0.003" rpy="0.0 0.0 0.0"/>
       <geometry>
-        <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_r_foot_rear_prt.stl" scale="0.001 0.001 0.001"/>
+        <box size="0.117 0.1 0.006"/>
       </geometry>
     </collision>
   </link>
@@ -316,7 +316,7 @@
   </link>
   <joint name="torso_roll" type="revolute">
     <origin xyz="0 0 0.0605" rpy="0 0 0"/>
-    <axis xyz="1.0 0.0 0.0"/>
+    <axis xyz="-1.0 0.0 0.0"/>
     <parent link="root_link"/>
     <child link="torso_1"/>
     <limit effort="50000" lower="-0.4014257279586958" upper="0.4014257279586958" velocity="120.0"/>
@@ -346,7 +346,7 @@
   </link>
   <joint name="torso_pitch" type="revolute">
     <origin xyz="0 0 0.07250000000000001" rpy="0 0 0"/>
-    <axis xyz="0.0 -1.0 0.0"/>
+    <axis xyz="0.0 1.0 0.0"/>
     <parent link="torso_1"/>
     <child link="torso_2"/>
     <limit effort="50000" lower="-0.3141592653589793" upper="0.7853981633974483" velocity="120.0"/>
@@ -406,7 +406,7 @@
   </link>
   <joint name="l_shoulder_pitch" type="revolute">
     <origin xyz="-0.020835100000000002 0.0397423 0.1052029" rpy="0.2532616237670649 -1.1982921379466467e-07 -0.26179930185680106"/>
-    <axis xyz="3.3231509266596455e-07 0.9999999783534343 1.4456511512603564e-07"/>
+    <axis xyz="3.323150926732427e-07 0.9999999783534343 1.445651151288363e-07"/>
     <parent link="chest"/>
     <child link="l_shoulder_1"/>
     <limit effort="50000" lower="-1.53588974175501" upper="0.22689280275926285" velocity="120.0"/>
@@ -414,12 +414,12 @@
   </joint>
   <link name="l_shoulder_2">
     <inertial>
-      <origin xyz="0.003293362025486902 0.0002563804707528261 -0.027865685757097337" rpy="-0.0715644949880594 -0.06196998841864678 -1.295357449823658"/>
+      <origin xyz="0.0032933620254868984 0.0002563804707528261 -0.027865685757097337" rpy="-0.07156449498805939 -0.06196998841864677 -1.2953574498236577"/>
       <mass value="0.322126"/>
       <inertia ixx="0.01" ixy="-3.5032e-05" ixz="-2.4436e-05" iyy="0.01" iyz="-5.52877e-05" izz="0.01"/>
     </inertial>
     <visual>
-      <origin xyz="0.02639309786919335 -0.14153112610649204 -0.007252397644602098" rpy="-0.0715644949880594 -0.06196998841864678 -1.295357449823658"/>
+      <origin xyz="0.026393097869193378 -0.14153112610649202 -0.0072523976446021024" rpy="-0.07156449498805939 -0.06196998841864677 -1.2953574498236577"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_l_shoulder_2_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
@@ -428,15 +428,15 @@
       </material>
     </visual>
     <collision>
-      <origin xyz="0.02639309786919335 -0.14153112610649204 -0.007252397644602098" rpy="-0.0715644949880594 -0.06196998841864678 -1.295357449823658"/>
+      <origin xyz="0.026393097869193378 -0.14153112610649202 -0.0072523976446021024" rpy="-0.07156449498805939 -0.06196998841864677 -1.2953574498236577"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_l_shoulder_2_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
     </collision>
   </link>
   <joint name="l_shoulder_roll" type="revolute">
-    <origin xyz="0.004992849332498911 0.11149983845605199 0.00026865023223796025" rpy="-0.17453296136691587 -0.05374759926346641 0"/>
-    <axis xyz="0.9999998262417685 2.0348252625938444e-07 -4.023022846062156e-08"/>
+    <origin xyz="0.004992849332498911 0.11149983845605199 0.00026865023223796025" rpy="-0.1745329613669159 -0.05374759926346645 0"/>
+    <axis xyz="0.9999998262417684 2.0348252649342313e-07 -4.0230228454118944e-08"/>
     <parent link="l_shoulder_1"/>
     <child link="l_shoulder_2"/>
     <limit effort="50000" lower="0.20943951023931956" upper="2.792526803190927" velocity="120.0"/>
@@ -444,12 +444,12 @@
   </joint>
   <link name="l_shoulder_3">
     <inertial>
-      <origin xyz="-4.416264022005789e-06 9.587749727052453e-05 -0.007326213473011087" rpy="-0.0715644949880594 -0.06196998841864678 -1.295357449823658"/>
+      <origin xyz="-4.41626402200232e-06 9.587749727052453e-05 -0.007326213473011087" rpy="-0.07156449498805939 -0.06196998841864677 -1.2953574498236577"/>
       <mass value="0.256128"/>
       <inertia ixx="0.01" ixy="-3.76336e-07" ixz="2.93189e-06" iyy="0.01" iyz="-3.52325e-06" izz="0.01"/>
     </inertial>
     <visual>
-      <origin xyz="0.03074323102617168 -0.1415314893638154 0.04704764107945962" rpy="-0.0715644949880594 -0.06196998841864678 -1.295357449823658"/>
+      <origin xyz="0.030743231026171713 -0.1415314893638154 0.047047641079459614" rpy="-0.07156449498805939 -0.06196998841864677 -1.2953574498236577"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_l_shoulder_3_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
@@ -458,26 +458,26 @@
       </material>
     </visual>
     <collision>
-      <origin xyz="0.03074323102617168 -0.1415314893638154 0.04704764107945962" rpy="-0.0715644949880594 -0.06196998841864678 -1.295357449823658"/>
+      <origin xyz="0.030743231026171713 -0.1415314893638154 0.047047641079459614" rpy="-0.07156449498805939 -0.06196998841864677 -1.2953574498236577"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_l_shoulder_3_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
     </collision>
   </link>
   <joint name="l_arm_ft_sensor" type="fixed">
-    <origin xyz="-0.004350133156978318 3.632573233602887e-07 -0.05430003872406172" rpy="0 0 0"/>
+    <origin xyz="-0.004350133156978329 3.632573233880443e-07 -0.05430003872406172" rpy="0 0 0"/>
     <parent link="l_shoulder_2"/>
     <child link="l_shoulder_3"/>
     <dynamics damping="0.1"/>
   </joint>
   <link name="l_upper_arm">
     <inertial>
-      <origin xyz="0.006736834478024374 -0.021799255751033003 -0.04590612014684704" rpy="-0.07156449487846149 -0.06196998844956256 -1.2953574498705325"/>
+      <origin xyz="0.006736834478024381 -0.02179925575103303 -0.04590612014684704" rpy="-0.07156449487846148 -0.06196998844956253 -1.295357449870532"/>
       <mass value="2.07167"/>
       <inertia ixx="0.0038606" ixy="0.000215061" ixz="-5.23479e-05" iyy="0.018139" iyz="0.000254564" izz="0.0161052"/>
     </inertial>
     <visual>
-      <origin xyz="0.030743149463830045 -0.14153106591257655 0.06954758245615261" rpy="-0.07156449487846149 -0.06196998844956256 -1.2953574498705325"/>
+      <origin xyz="0.03074314946383008 -0.14153106591257655 0.06954758245615261" rpy="-0.07156449487846148 -0.06196998844956253 -1.295357449870532"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_l_upperarm_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
@@ -486,15 +486,15 @@
       </material>
     </visual>
     <collision>
-      <origin xyz="0.030743149463830045 -0.14153106591257655 0.06954758245615261" rpy="-0.07156449487846149 -0.06196998844956256 -1.2953574498705325"/>
+      <origin xyz="0.03074314946383008 -0.14153106591257655 0.06954758245615261" rpy="-0.07156449487846148 -0.06196998844956253 -1.295357449870532"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_l_upperarm_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
     </collision>
   </link>
   <joint name="l_shoulder_yaw" type="revolute">
-    <origin xyz="0 -4.2345248449371375e-07 -0.022499941373124423" rpy="0 0 0"/>
-    <axis xyz="-2.2276474049102235e-08 4.393786247302041e-08 1.0000002566182309"/>
+    <origin xyz="0 -4.2345248449371375e-07 -0.02249994137312443" rpy="0 0 0"/>
+    <axis xyz="-2.227647404040578e-08 4.393786245948661e-08 1.0000002566182309"/>
     <parent link="l_shoulder_3"/>
     <child link="l_upper_arm"/>
     <limit effort="50000" lower="-0.8726646259971648" upper="1.3962634015954636" velocity="120.0"/>
@@ -502,12 +502,12 @@
   </joint>
   <link name="l_elbow_1">
     <inertial>
-      <origin xyz="-0.010241946078109288 0.0018224069895867168 -0.015089620258555453" rpy="-0.07156449487846121 -0.06196998844956255 -1.2953574498905756"/>
+      <origin xyz="-0.010241946078109292 0.0018224069895867168 -0.01508962025855548" rpy="-0.07156449487846119 -0.0619699884495625 -1.2953574498905749"/>
       <mass value="0.230254"/>
       <inertia ixx="0.01" ixy="7.51175e-07" ixz="-9.8725e-06" iyy="0.01" iyz="-1.03338e-05" izz="0.01"/>
     </inertial>
     <visual>
-      <origin xyz="0.020743219226073948 -0.14003141848976391 0.19324791999373644" rpy="-0.07156449487846121 -0.06196998844956255 -1.2953574498905756"/>
+      <origin xyz="0.02074321922607398 -0.14003141848976391 0.19324791999373642" rpy="-0.07156449487846119 -0.0619699884495625 -1.2953574498905749"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_l_elbow_1_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
@@ -516,7 +516,7 @@
       </material>
     </visual>
     <collision>
-      <origin xyz="0.020743219226073948 -0.14003141848976391 0.19324791999373644" rpy="-0.07156449487846121 -0.06196998844956255 -1.2953574498905756"/>
+      <origin xyz="0.02074321922607398 -0.14003141848976391 0.19324791999373642" rpy="-0.07156449487846119 -0.0619699884495625 -1.2953574498905749"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_l_elbow_1_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
@@ -524,7 +524,7 @@
   </link>
   <joint name="l_elbow" type="revolute">
     <origin xyz="0.009999930234919404 -0.001499647423428896 -0.12370033753758375" rpy="0 0 0"/>
-    <axis xyz="-8.811824174789715e-08 -1.0000002733514628 4.404736120498342e-08"/>
+    <axis xyz="-8.811824154381087e-08 -1.0000002733514628 4.4047361189014644e-08"/>
     <parent link="l_upper_arm"/>
     <child link="l_elbow_1"/>
     <limit effort="50000" lower="-0.05235987755982989" upper="1.3089969389957472" velocity="120.0"/>
@@ -532,12 +532,12 @@
   </joint>
   <link name="l_forearm">
     <inertial>
-      <origin xyz="-0.0062698741022980675 -4.68551245806359e-05 -0.030648347755607208" rpy="-0.0715644949880594 -0.06196998841864678 -1.295357449823658"/>
+      <origin xyz="-0.006269874102298061 -4.68551245806359e-05 -0.030648347755607153" rpy="-0.07156449498805939 -0.06196998841864677 -1.2953574498236577"/>
       <mass value="0.90793"/>
       <inertia ixx="0.00131822" ixy="-7.39737e-07" ixz="-2.69292e-05" iyy="0.00128448" iyz="0.000108551" izz="0.000646975"/>
     </inertial>
     <visual>
-      <origin xyz="0.035743089887937445 -0.14203106199153737 0.2485479835726478" rpy="-0.0715644949880594 -0.06196998841864678 -1.295357449823658"/>
+      <origin xyz="0.03574308988793748 -0.14203106199153737 0.2485479835726478" rpy="-0.07156449498805939 -0.06196998841864677 -1.2953574498236577"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_l_forearm_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
@@ -546,15 +546,15 @@
       </material>
     </visual>
     <collision>
-      <origin xyz="0.035743089887937445 -0.14203106199153737 0.2485479835726478" rpy="-0.0715644949880594 -0.06196998841864678 -1.295357449823658"/>
+      <origin xyz="0.03574308988793748 -0.14203106199153737 0.2485479835726478" rpy="-0.07156449498805939 -0.06196998841864677 -1.2953574498236577"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_l_forearm_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
     </collision>
   </link>
   <joint name="l_wrist_prosup" type="revolute">
-    <origin xyz="-0.0149998706541777 0.0019996435036348514 -0.0553000635824798" rpy="0 0 0"/>
-    <axis xyz="-2.2276474049102235e-08 4.393786247302041e-08 1.0000002566182309"/>
+    <origin xyz="-0.014999870654177704 0.0019996435036348514 -0.055300063582479825" rpy="0 0 0"/>
+    <axis xyz="-2.227647404040578e-08 4.393786245948661e-08 1.0000002566182309"/>
     <parent link="l_elbow_1"/>
     <child link="l_forearm"/>
     <limit effort="50000" lower="-1.53588974175501" upper="1.53588974175501" velocity="120.0"/>
@@ -562,12 +562,12 @@
   </joint>
   <link name="l_wrist_1">
     <inertial>
-      <origin xyz="-0.014390150453737368 -0.0005083960996520864 -9.893035257668714e-05" rpy="-0.0715644945595831 -0.0619699885395119 -1.2953574498501936"/>
+      <origin xyz="-0.014390150453737368 -0.0005083960996520587 -9.893035257668714e-05" rpy="-0.07156449455958307 -0.06196998853951186 -1.2953574498501934"/>
       <mass value="0.0344005"/>
       <inertia ixx="0.01" ixy="-9.19776e-07" ixz="-1.4756e-08" iyy="0.01" iyz="-1.11493e-07" izz="0.01"/>
     </inertial>
     <visual>
-      <origin xyz="0.01914307389058231 -0.14403795720942245 0.3560476247256178" rpy="-0.0715644945595831 -0.0619699885395119 -1.2953574498501936"/>
+      <origin xyz="0.019143073890582344 -0.14403795720942242 0.3560476247256178" rpy="-0.07156449455958307 -0.06196998853951186 -1.2953574498501934"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_l_wrist_1_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
@@ -576,15 +576,15 @@
       </material>
     </visual>
     <collision>
-      <origin xyz="0.01914307389058231 -0.14403795720942245 0.3560476247256178" rpy="-0.0715644945595831 -0.0619699885395119 -1.2953574498501936"/>
+      <origin xyz="0.019143073890582344 -0.14403795720942242 0.3560476247256178" rpy="-0.07156449455958307 -0.06196998853951186 -1.2953574498501934"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_l_wrist_1_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
     </collision>
   </link>
   <joint name="l_wrist_pitch" type="revolute">
-    <origin xyz="0.016600015934286182 0.0020068952180725685 -0.10749964114254112" rpy="0 0 0"/>
-    <axis xyz="0.9999998262417685 2.0348252625938444e-07 -4.023022846062156e-08"/>
+    <origin xyz="0.016600015934286182 0.002006895218072541 -0.10749964114254112" rpy="0 0 0"/>
+    <axis xyz="0.9999998262417684 2.0348252649342313e-07 -4.0230228454118944e-08"/>
     <parent link="l_forearm"/>
     <child link="l_wrist_1"/>
     <limit effort="50000" lower="-1.2217304763960306" upper="0.4363323129985824" velocity="120.0"/>
@@ -592,12 +592,12 @@
   </joint>
   <link name="l_hand">
     <inertial>
-      <origin xyz="0.004786523181334726 -0.0061337378723908265 -0.0672770617484692" rpy="-0.07156449455958207 -0.06196998853951186 -1.2953574498501934"/>
+      <origin xyz="0.004786523181334722 -0.0061337378723908265 -0.0672770617484692" rpy="-0.07156449455958207 -0.061969988539511835 -1.295357449850193"/>
       <mass value="0.263541"/>
       <inertia ixx="0.01" ixy="-4.77094e-05" ixz="-1.03141e-05" iyy="0.01" iyz="1.19895e-05" izz="0.01"/>
     </inertial>
     <visual>
-      <origin xyz="0.0354431956353115 -0.13703856452785146 0.3560477032643494" rpy="-0.07156449455958207 -0.06196998853951186 -1.2953574498501934"/>
+      <origin xyz="0.035443195635311536 -0.13703856452785146 0.3560477032643494" rpy="-0.07156449455958207 -0.061969988539511835 -1.295357449850193"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_l_hand_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
@@ -606,15 +606,15 @@
       </material>
     </visual>
     <collision>
-      <origin xyz="0.0354431956353115 -0.13703856452785146 0.3560477032643494" rpy="-0.07156449455958207 -0.06196998853951186 -1.2953574498501934"/>
+      <origin xyz="0.035443195635311536 -0.13703856452785146 0.3560477032643494" rpy="-0.07156449455958207 -0.061969988539511835 -1.295357449850193"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_l_hand_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
     </collision>
   </link>
   <joint name="l_wrist_yaw" type="revolute">
-    <origin xyz="-0.016300121744729387 -0.006999392681571209 0" rpy="0 0 0"/>
-    <axis xyz="-8.811824174789715e-08 -1.0000002733514628 4.404736120498342e-08"/>
+    <origin xyz="-0.016300121744729387 -0.0069993926815711816 0" rpy="0 0 0"/>
+    <axis xyz="-8.811824154381087e-08 -1.0000002733514628 4.4047361189014644e-08"/>
     <parent link="l_wrist_1"/>
     <child link="l_hand"/>
     <limit effort="50000" lower="-0.2617993877991494" upper="0.6108652381980153" velocity="120.0"/>
@@ -644,7 +644,7 @@
   </link>
   <joint name="r_shoulder_pitch" type="revolute">
     <origin xyz="-0.020835100000000002 -0.0397423 0.1052029" rpy="-0.25326164431680925 2.3014774394592079e-07 0.26179970938509745"/>
-    <axis xyz="1.499044390098182e-07 0.9999999783534812 -1.2401537036323317e-07"/>
+    <axis xyz="1.499044390054707e-07 0.9999999783534812 -1.2401537036188515e-07"/>
     <parent link="chest"/>
     <child link="r_shoulder_1"/>
     <limit effort="50000" lower="-1.53588974175501" upper="0.22689280275926285" velocity="120.0"/>
@@ -652,12 +652,12 @@
   </joint>
   <link name="r_shoulder_2">
     <inertial>
-      <origin xyz="0.000711844151437005 -0.00028997131506725005 -0.027865575116629175" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
+      <origin xyz="0.0007118441514370015 -0.0002899713150672778 -0.027865575116629175" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
       <mass value="0.322126"/>
       <inertia ixx="0.01" ixy="3.44034e-05" ixz="2.39129e-05" iyy="0.01" iyz="-5.45925e-05" izz="0.01"/>
     </inertial>
     <visual>
-      <origin xyz="0.02389326106684631 0.14153171676199228 -0.007252350207672734" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
+      <origin xyz="0.023893261066846305 0.14153171676199225 -0.007252350207672735" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_r_shoulder_2_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
@@ -666,15 +666,15 @@
       </material>
     </visual>
     <collision>
-      <origin xyz="0.02389326106684631 0.14153171676199228 -0.007252350207672734" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
+      <origin xyz="0.023893261066846305 0.14153171676199225 -0.007252350207672735" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_r_shoulder_2_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
     </collision>
   </link>
   <joint name="r_shoulder_roll" type="revolute">
-    <origin xyz="0.007489008185044645 -0.11150041046657978 0.0004027858966975245" rpy="0.17453306234069352 -0.05374779257913059 -6.132332688804098e-07"/>
-    <axis xyz="-0.9999998262417886 4.210501434612913e-08 2.4630481182686846e-08"/>
+    <origin xyz="0.007489008185044645 -0.11150041046657978 0.0004027858966975245" rpy="0.17453306234069357 -0.053747792579130614 -6.132332688812785e-07"/>
+    <axis xyz="-0.9999998262417886 4.2105014356162707e-08 2.4630481189336084e-08"/>
     <parent link="r_shoulder_1"/>
     <child link="r_shoulder_2"/>
     <limit effort="50000" lower="0.20943951023931956" upper="2.792526803190927" velocity="120.0"/>
@@ -682,12 +682,12 @@
   </joint>
   <link name="r_shoulder_3">
     <inertial>
-      <origin xyz="-8.928496484206933e-05 -3.376971940446949e-05 -0.007594500252887257" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
+      <origin xyz="-8.928496484206933e-05 -3.376971940446949e-05 -0.00759450025288725" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
       <mass value="0.258699"/>
       <inertia ixx="0.01" ixy="5.87638e-07" ixz="-2.8754e-06" iyy="0.01" iyz="-3.42816e-06" izz="0.01"/>
     </inertial>
     <visual>
-      <origin xyz="0.030743207452236184 0.1415314977120317 0.047047631370218244" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
+      <origin xyz="0.030743207452236184 0.1415314977120317 0.04704763137021825" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_r_shoulder_3_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
@@ -696,26 +696,26 @@
       </material>
     </visual>
     <collision>
-      <origin xyz="0.030743207452236184 0.1415314977120317 0.047047631370218244" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
+      <origin xyz="0.030743207452236184 0.1415314977120317 0.04704763137021825" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_r_shoulder_3_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
     </collision>
   </link>
   <joint name="r_arm_ft_sensor" type="fixed">
-    <origin xyz="-0.006849946385389872 2.1904996058697002e-07 -0.05429998157789099" rpy="0 0 0"/>
+    <origin xyz="-0.006849946385389875 2.1904996055921444e-07 -0.05429998157789099" rpy="0 0 0"/>
     <parent link="r_shoulder_2"/>
     <child link="r_shoulder_3"/>
     <dynamics damping="0.1"/>
   </joint>
   <link name="r_upper_arm">
     <inertial>
-      <origin xyz="-4.827495476422694e-06 -0.001943485876269635 -0.04747357932833829" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
+      <origin xyz="-4.8274954764261635e-06 -0.001943485876269635 -0.04747357932833829" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
       <mass value="2.06334"/>
       <inertia ixx="0.00382639" ixy="-5.03894e-06" ixz="0.000205684" iyy="0.00368146" iyz="0.000261155" izz="0.00167002"/>
     </inertial>
     <visual>
-      <origin xyz="0.030743125543818535 0.141531075879301 0.06954757274337192" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
+      <origin xyz="0.03074312554381853 0.141531075879301 0.06954757274337192" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_r_upperarm_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
@@ -724,15 +724,15 @@
       </material>
     </visual>
     <collision>
-      <origin xyz="0.030743125543818535 0.141531075879301 0.06954757274337192" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
+      <origin xyz="0.03074312554381853 0.141531075879301 0.06954757274337192" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_r_upperarm_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
     </collision>
   </link>
   <joint name="r_shoulder_yaw" type="revolute">
-    <origin xyz="0 4.2183273077367467e-07 -0.02249994137315358" rpy="0 0 0"/>
-    <axis xyz="-3.787621790776764e-08 2.805197706023475e-08 1.0000002566182309"/>
+    <origin xyz="0 4.2183273077367467e-07 -0.022499941373153573" rpy="0 0 0"/>
+    <axis xyz="-3.787621790513063e-08 2.805197706068352e-08 1.0000002566182309"/>
     <parent link="r_shoulder_3"/>
     <child link="r_upper_arm"/>
     <limit effort="50000" lower="-0.8726646259971648" upper="1.3962634015954636" velocity="120.0"/>
@@ -740,12 +740,12 @@
   </joint>
   <link name="r_elbow_1">
     <inertial>
-      <origin xyz="-0.011624818009452066 -0.0018028159470199479 0.022300137248866647" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
+      <origin xyz="-0.011624818009452066 -0.00180281594701992 0.022300137248866647" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
       <mass value="0.397478"/>
       <inertia ixx="0.01" ixy="5.80993e-06" ixz="5.27143e-05" iyy="0.01" iyz="6.49755e-05" izz="0.01"/>
     </inertial>
     <visual>
-      <origin xyz="0.020743193621214625 0.14003143574727353 0.19324791023291843" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
+      <origin xyz="0.020743193621214625 0.14003143574727356 0.19324791023291843" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_r_elbow_1_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
@@ -754,15 +754,15 @@
       </material>
     </visual>
     <collision>
-      <origin xyz="0.020743193621214625 0.14003143574727353 0.19324791023291843" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
+      <origin xyz="0.020743193621214625 0.14003143574727356 0.19324791023291843" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_r_elbow_1_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
     </collision>
   </link>
   <joint name="r_elbow" type="revolute">
-    <origin xyz="0.00999993192260382 0.0014996401320274133 -0.1237003374895465" rpy="0 0 0"/>
-    <axis xyz="2.4949582807394525e-07 -1.0000002733514362 2.7942487010105877e-08"/>
+    <origin xyz="0.009999931922603816 0.0014996401320274133 -0.1237003374895465" rpy="0 0 0"/>
+    <axis xyz="2.4949582809984105e-07 -1.0000002733514362 2.7942487004772865e-08"/>
     <parent link="r_upper_arm"/>
     <child link="r_elbow_1"/>
     <limit effort="50000" lower="-0.05235987755982989" upper="1.3089969389957472" velocity="120.0"/>
@@ -791,8 +791,8 @@
     </collision>
   </link>
   <joint name="r_wrist_prosup" type="revolute">
-    <origin xyz="-0.014999869468812953 -0.0019996499053189787 -0.05530006367251966" rpy="0 0 0"/>
-    <axis xyz="-3.787621790776764e-08 2.805197706023475e-08 1.0000002566182309"/>
+    <origin xyz="-0.014999869468812953 -0.001999649905318951 -0.05530006367251966" rpy="0 0 0"/>
+    <axis xyz="-3.787621790513063e-08 2.805197706068352e-08 1.0000002566182309"/>
     <parent link="r_elbow_1"/>
     <child link="r_forearm"/>
     <limit effort="50000" lower="-1.53588974175501" upper="1.53588974175501" velocity="120.0"/>
@@ -800,12 +800,12 @@
   </joint>
   <link name="r_wrist_1">
     <inertial>
-      <origin xyz="-0.014389827079018234 0.0005092753390521787 -9.78731576423697e-05" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
+      <origin xyz="-0.014389827079018234 0.0005092753390522065 -9.787315764231419e-05" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
       <mass value="0.0344005"/>
       <inertia ixx="0.01" ixy="9.19776e-07" ixz="1.47559e-08" iyy="0.01" iyz="-1.11493e-07" izz="0.01"/>
     </inertial>
     <visual>
-      <origin xyz="0.01944310973607908 0.14403852854340446 0.35604805361754444" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
+      <origin xyz="0.01944310973607908 0.1440385285434045 0.3560480536175445" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_r_wrist_1_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
@@ -814,7 +814,7 @@
       </material>
     </visual>
     <collision>
-      <origin xyz="0.01944310973607908 0.14403852854340446 0.35604805361754444" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
+      <origin xyz="0.01944310973607908 0.1440385285434045 0.3560480536175445" rpy="-0.07156446033079357 0.06196992338505823 -1.8462350402422587"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_r_wrist_1_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
@@ -822,7 +822,7 @@
   </link>
   <joint name="r_wrist_pitch" type="revolute">
     <origin xyz="0.016299953353948633 -0.002007442890811839 -0.10750007971210637" rpy="0 0 0"/>
-    <axis xyz="0.9999998262417886 -4.210501434612913e-08 -2.4630481182686846e-08"/>
+    <axis xyz="0.9999998262417886 -4.2105014356162707e-08 -2.4630481189336084e-08"/>
     <parent link="r_forearm"/>
     <child link="r_wrist_1"/>
     <limit effort="50000" lower="-1.2217304763960306" upper="0.4363323129985824" velocity="120.0"/>
@@ -830,12 +830,12 @@
   </joint>
   <link name="r_hand">
     <inertial>
-      <origin xyz="0.004828314385249669 0.00615667237451184 -0.0671378028889843" rpy="-0.07156446894052497 0.061969920956429156 -1.8462350407754615"/>
+      <origin xyz="0.004828314385249662 0.00615667237451184 -0.0671378028889843" rpy="-0.07156446894052497 0.061969920956429156 -1.8462350407754615"/>
       <mass value="0.264229"/>
       <inertia ixx="0.01" ixy="4.7873e-05" ixz="1.1114e-05" iyy="0.01" iyz="1.14469e-05" izz="0.01"/>
     </inertial>
     <visual>
-      <origin xyz="0.03574323578978202 0.1370391384923153 0.35604813259525964" rpy="-0.07156446894052497 0.061969920956429156 -1.8462350407754615"/>
+      <origin xyz="0.035743235789782016 0.1370391384923153 0.35604813259525964" rpy="-0.07156446894052497 0.061969920956429156 -1.8462350407754615"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_r_hand_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
@@ -844,22 +844,15 @@
       </material>
     </visual>
     <collision>
-      <origin xyz="0.03574323578978202 0.1370391384923153 0.35604813259525964" rpy="-0.07156446894052497 0.061969920956429156 -1.8462350407754615"/>
+      <origin xyz="0.035743235789782016 0.1370391384923153 0.35604813259525964" rpy="-0.07156446894052497 0.061969920956429156 -1.8462350407754615"/>
       <geometry>
         <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_r_hand_prt.stl" scale="0.001 0.001 0.001"/>
       </geometry>
     </collision>
   </link>
-  <link name="imu_frame"/>
-  <joint name="imu_frame_fixed_joint" type="fixed">
-    <origin xyz="0.009500000190735 0.133444 0.009299999999999996" rpy="-1.5707963267948961 1.570796326794896 0"/>
-    <parent link="head"/>
-    <child link="imu_frame"/>
-    <dynamics damping="0.1"/>
-  </joint>  
   <joint name="r_wrist_yaw" type="continuous">
-    <origin xyz="-0.016300122874272734 0.006999390051091792 0" rpy="0 0 0"/>
-    <axis xyz="-2.4949582807394525e-07 1.0000002733514362 -2.7942487010105877e-08"/>
+    <origin xyz="-0.016300122874272734 0.0069993900510918194 0" rpy="0 0 0"/>
+    <axis xyz="-2.4949582809984105e-07 1.0000002733514362 -2.7942487004772865e-08"/>
     <parent link="r_wrist_1"/>
     <child link="r_hand"/>
     <dynamics damping="0.1"/>
@@ -1134,7 +1127,7 @@
     <inertial>
       <origin xyz="-0.0049152000000000015 -0.0022939000000000154 -0.018371000000000026" rpy="0 0 -1.5707963267948966"/>
       <mass value="1.80595"/>
-      <inertia ixx="0.00329109" ixy="0.000115135" ixz="0.000232789" iyy="0.00097845" iyz="-0.000112215" izz="0.00298837"/>
+      <inertia ixx="0.01" ixy="0.000115135" ixz="0.000232789" iyy="0.01" iyz="-0.000112215" izz="0.01"/>
     </inertial>
     <visual>
       <origin xyz="-0.026049999999999997 -0.08294999999999998 0.8456470000000004" rpy="0 0 -1.5707963267948966"/>
@@ -1154,7 +1147,7 @@
   </link>
   <joint name="l_ankle_pitch" type="revolute">
     <origin xyz="0 -0.0005000000000000004 -0.255989" rpy="0 0 0"/>
-    <axis xyz="0.0 -1.0 0.0"/>
+    <axis xyz="0.0 1.0 0.0"/>
     <parent link="l_lower_leg"/>
     <child link="l_ankle_1"/>
     <limit effort="50000" lower="-0.7853981633974483" upper="0.7853981633974483" velocity="120.0"/>
@@ -1164,7 +1157,7 @@
     <inertial>
       <origin xyz="0.0395516 -8.099999999999774e-05 -0.025341999999999976" rpy="0 0 -1.5707963267948966"/>
       <mass value="1.06976"/>
-      <inertia ixx="0.00510275" ixy="4.67014e-06" ixz="-6.74116e-07" iyy="0.000995114" iyz="0.000903931" izz="0.00458166"/>
+      <inertia ixx="0.01" ixy="4.67014e-06" ixz="-6.74116e-07" iyy="0.01" iyz="0.000903931" izz="0.01"/>
     </inertial>
     <visual>
       <origin xyz="0.026700000000000015 -0.06475000000001183 0.8806469999999995" rpy="0 0 -1.5707963267948966"/>
@@ -1193,8 +1186,8 @@
   <link name="l_foot_front">
     <inertial>
       <origin xyz="-0.00024399999999999422 9.139999999999149e-05 0.003298999999999941" rpy="0 0 -1.5707963267948966"/>
-      <mass value="0.723938"/>
-      <inertia ixx="0.000770637" ixy="7.67083e-07" ixz="-4.91844e-07" iyy="0.000568064" iyz="-5.214e-07" izz="0.001333"/>
+      <mass value="0.18507432"/>
+      <inertia ixx="0.01" ixy="1.961043136408919e-07" ixz="-1.257396266615097e-07" iyy="0.01" iyz="-1.3329560051827644e-07" izz="0.01"/>
     </inertial>
     <visual>
       <origin xyz="-0.10854999999999998 -0.06475000000001183 0.9464469999999995" rpy="0 0 -1.5707963267948966"/>
@@ -1206,9 +1199,9 @@
       </material>
     </visual>
     <collision>
-      <origin xyz="-0.10854999999999998 -0.06475000000001183 0.9464469999999995" rpy="0 0 -1.5707963267948966"/>
+      <origin xyz="0.0 0.0 0.003" rpy="0.0 0.0 0.0"/>
       <geometry>
-        <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_l_foot_front_prt.stl" scale="0.001 0.001 0.001"/>
+        <box size="0.117 0.1 0.006"/>
       </geometry>
     </collision>
   </link>
@@ -1221,8 +1214,8 @@
   <link name="l_foot_rear">
     <inertial>
       <origin xyz="0.00020089999999999865 -3.260000000000762e-05 0.003339000000000092" rpy="0 0 -1.5707963267948966"/>
-      <mass value="0.728095"/>
-      <inertia ixx="0.000769193" ixy="-1.67298e-06" ixz="-7.88069e-07" iyy="0.000565835" iyz="-5.1234e-07" izz="0.00132907"/>
+      <mass value="0.18460842"/>
+      <inertia ixx="0.01" ixy="-4.2418392447633885e-07" ixz="-1.9981482216054221e-07" iyy="0.01" iyz="-1.2990375967806395e-07" izz="0.01"/>
     </inertial>
     <visual>
       <origin xyz="0.010700000000000012 -0.06475000000001183 0.9464469999999995" rpy="0 0 -1.5707963267948966"/>
@@ -1234,9 +1227,9 @@
       </material>
     </visual>
     <collision>
-      <origin xyz="0.010700000000000012 -0.06475000000001183 0.9464469999999995" rpy="0 0 -1.5707963267948966"/>
+      <origin xyz="0.0 0.0 0.003" rpy="0.0 0.0 0.0"/>
       <geometry>
-        <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_l_foot_rear_prt.stl" scale="0.001 0.001 0.001"/>
+        <box size="0.117 0.1 0.006"/>
       </geometry>
     </collision>
   </link>
@@ -1663,25 +1656,6 @@
     </force_torque>
   </sensor>
   <gazebo>
-    <pose>0.0 0.0 0.8 0.0 0.0 0.0</pose>
-  </gazebo>
-  <gazebo reference="head">
-    <sensor name="head_imu_acc_1x1" type="imu">
-      <always_on>1</always_on>
-      <update_rate>100</update_rate>
-      <pose>0.0 0.0 0.0 0.0 0.0 0.0</pose>
-        <plugin name="iCub_yarp_gazebo_plugin_IMU" filename="libgazebo_yarp_imu.so">
-            <yarpConfigurationFile>model://iCub/conf_icub3/gazebo_icub_inertial.ini</yarpConfigurationFile>
-        </plugin>
-    </sensor>
-  </gazebo>
-  <sensor name="head_imu_acc_1x1" type="accelerometer">
-    <parent link="head"/>
-    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
-  </sensor>
-  <gazebo>
-    <plugin name="controlboard_wholebody_ros_nohands" filename="libgazebo_yarp_controlboard.so">
-        <yarpConfigurationFile>model://iCub/conf_icub3/gazebo_icub_wholebody_ros_nohands.ini</yarpConfigurationFile>
-    </plugin>
+    <pose>0.0 0.0 0.64 0.0 0.0 0.0</pose>
   </gazebo>
 </robot>
```

cc @traversaro @prashanthr05 @S-Dafarra

**Comment by prashanthr05** [comment]
The recent changes in the `icub-models-generator` for `iCubV3` models have not been accounted for. 
I think a rebase would make things complicated.
As suggested by @traversaro, https://github.com/dic-iit/component_ANA-Avatar-XPRIZE/issues/82, the right way to do this is through icub-models-generator and since the URDF is constantly evolving, rebasing everytime would be painful.

Instead we might have to consider including the URDF that's constantly evolving along with the custom modifications we would like to have on the model.

**Comment by prashanthr05** [comment]
> Instead we might have to consider including the URDF that's constantly evolving along with the custom modifications we would like to have on the model.

@GiulioRomualdi  I had to investigate it but currently I am working on the iFeel sprint. 
I am not sure if I will be able to do that at the quickest.

**Comment by GiulioRomualdi** [comment]
> I think a rebase would make things complicated.

I agree but I hope that this current situation is only temporary

**Comment by prashanthr05** [comment]
> > I think a rebase would make things complicated.
> 
> I agree but I hope that this current situation is only temporary

if we are successful with the icub-models-generator workflow, I suppose this situation will be temporary.

**Comment by S-Dafarra** [comment]
I did not understand two things:
- which version is the correct one in terms of axis direction?
- what are the difficulties in rebasing?

**Comment by prashanthr05** [comment]
> I did not understand two things:
> 
> * which version is the correct one in terms of axis direction?

The latest `icub-models` in remote `origin` would be the correct, since it accounts for the fixes introduced by @GiulioRomualdi 

> * what are the difficulties in rebasing?

It could be error-prone in case of merge conflicts. And we might have to do it repeatedly to be aligned with continuous changes in the remote.

**Comment by S-Dafarra** [comment]
> It could be error-prone in case of merge conflicts. And we might have to do it repeatedly to be aligned with continuous changes in the remote.

Where possible merge conflict could occur?

**Comment by prashanthr05** [comment]
> > It could be error-prone in case of merge conflicts. And we might have to do it repeatedly to be aligned with continuous changes in the remote.
> 
> Where possible merge conflict could occur?

I am speculating even in the [above scenario](https://github.com/dic-iit/icub-models/issues/1#issue-743723324), trying to rebase would result in us resolving some changes manually. 

and In case we happen to resolve any conflicts manually, there are possibilities of
- introducing/removing xml tags at wrong places
- due to the long floating point numbers, it also involves paying too much attention to those details along with also much context switching

**Comment by S-Dafarra** [comment]
> > > It could be error-prone in case of merge conflicts. And we might have to do it repeatedly to be aligned with continuous changes in the remote.
> > 
> > 
> > Where possible merge conflict could occur?
> 
> I am speculating even in the [above scenario](https://github.com/dic-iit/icub-models/issues/1#issue-743723324), trying to rebase would result in us resolving some changes manually.
> 
> and In case we happen to resolve any conflicts manually, there are possibilities of
> 
> * introducing/removing xml tags at wrong places
> * due to the long floating point numbers, it also involves paying too much attention to those details along with also much context switching

I see! But in this specific case, I guess that the only possible sources of conflicts are related to manual modifications in the ``urdf``. I suspect that block additions (like https://github.com/dic-iit/icub-models/blob/20d034c547aae832c7f3db3c1f53398dc10fac76/iCub/robots/iCubGazeboV3/model.urdf#L1678-L1686) should be easy to handle by ``git``. For the all modifications you mentioned, we can simply use the ``remote`` ones, right?

**Comment by S-Dafarra** [comment]
In any case, @GiulioRomualdi have you tried to do the rebase locally?

**Comment by GiulioRomualdi** [comment]
> In any case, @GiulioRomualdi have you tried to do the rebase locally?

yes, we cannot rebase as it is. if we do a `git rebase origin/devel` we lose all the modifications related to the simplified contact shape for the feet. 

This 
```patch
-      <origin xyz="0.010700000000000012 -0.06475000000001183 0.9464469999999995" rpy="0 0 -1.5707963267948966"/>
+      <origin xyz="0.0 0.0 0.003" rpy="0.0 0.0 0.0"/>
       <geometry>
-        <mesh filename="package://iCub/meshes/simmechanics/sim_icub3_l_foot_rear_prt.stl" scale="0.001 0.001 0.001"/>
+        <box size="0.117 0.1 0.006"/>
```

Edit: I'm a bit confused :confused: Could you tell me which are the lines required to have whole-body dynamics running?

**Comment by GiulioRomualdi** [comment]
> Edit: I'm a bit confused :confused:. Could you tell me which are the lines required to have whole-body dynamics running?

Well @GiulioRomualdi, you may check the history of the commits:
- acff976074ad84262d92151df0a0a6c5dc3502e2
- df289b6039bcafb005365d8aafcbc80d0912a66b
- acff976074ad84262d92151df0a0a6c5dc3502e2

**Comment by S-Dafarra** [comment]
> yes, we cannot rebase as it is. if we do a `git rebase origin/devel` we lose all the modifications related to the simplified contact shape for the feet.

If I am not wrong, that feature got merged into ``icub-models-generator`` (https://github.com/robotology/icub-models-generator/pull/165) and arrived to ``devel`` with https://github.com/robotology/icub-models/commit/f95aa8759ed1d9bf8fb4b32d908b549573aa658c :thinking:

**Comment by prashanthr05** [comment]
> If I am not wrong, that feature got merged into icub-models-generator (robotology/icub-models-generator#165) and arrived to devel with robotology@f95aa87 thinking

my bad, I pointed @GiulioRomualdi  towards the `master` branch. While the changes are being merged into `devel`.  I confirm rebasing against `devel` does not result in any conflicts.

> Could you tell me which are the lines required to have whole-body dynamics running?

The lines  with + are required. 

The running of  whole-body-dynamics is independent of the changes in these lines.
These simplified collision geometries are useful to have better contacts at the feet.

**Comment by GiulioRomualdi** [comment]
I'm trying to rebase on my PC. Unfortunately, I'm not able to run whole-body dynamics because of https://github.com/robotology/whole-body-estimators/issues/91

**Comment by S-Dafarra** [comment]
> I'm trying to rebase on my PC. Unfortunately, I'm not able to run whole-body dynamics because of [robotology/whole-body-estimators#91](https://github.com/robotology/whole-body-estimators/issues/91)

Is this problem solved?

**Comment by prashanthr05** [comment]
I suppose, yes.

**Comment by GiulioRomualdi** [comment]
Yes I rebased the model locally on my pc. How shall we proceed? Do we open a PR from origin devel to this fork?

**Comment by S-Dafarra** [comment]
> Yes I rebased the model locally on my pc. How shall we proceed? Do we open a PR from origin devel to this fork?

I guess we can start aligning this fork with your local version. Then, I guess that the file additions can be merged directly. For the modifications to the URDF we should open a PR on ``icub-models-generator`` to avoid losing them as soon as the URDF is regenerated.

**Comment by prashanthr05** [comment]
I have rebased over the latest devel and force-pushed the branch. In the end, the only diff from the devel must be,

## iCubGazeboV3
- addition of all joints ROS wrapper
- commenting out Bosch IMU and using the XSense IMU in the head (params from iCubGenova01)

## iCubGenova09
- commenting out Bosch IMU and using the XSense IMU in the head (params from iCubGenova01)

**Comment by prashanthr05** [comment]
We can close this issue?
